client/src/components/ColorList.js

```diff
@@ -1,40 +1,70 @@
-import React, { useState } from "react";
-import axios from "axios";
+import React, { useState, useEffect } from "react";
+import axiosWithAuth from "../utils/axiosWithAuth";
 
 const initialColor = {
   color: "",
   code: { hex: "" }
 };
-
+const token = localStorage.getItem(token);
+let id;
 const ColorList = ({ colors, updateColors }) => {
   console.log(colors);
   const [editing, setEditing] = useState(false);
   const [colorToEdit, setColorToEdit] = useState(initialColor);
+  const [addColor, setAddColor] = useState(initialColor);
 
   const editColor = color => {
     setEditing(true);
     setColorToEdit(color);
   };
-
+  useEffect(() => {
+    if (colorToEdit.id != null) id = colorToEdit.id;
+  }, [editing]);
   const saveEdit = e => {
-    e.preventDefault();
     // Make a put request to save your updated color
     // think about where will you get the id from...
     // where is is saved right now?
+    axiosWithAuth()
+      .put(`/api/colors/${id}`, colorToEdit)
+      .then(res => {
+        console.log(res.data);
+        updateColors([...res.data])
+      })
+      .catch(err => console.log(err.response));
+      
   };
 
   const deleteColor = color => {
     // make a delete request to delete this color
+    axiosWithAuth()
+      .delete(`/api/colors/${color.id}`,token)
```

Comment:
Because our `axiosWithAuth()` uses the token from localStorage, we don't need to use it here '' '
```
.delete(`/api/colors/${color.id}`)
```